chromadb/db/duckdb.py

```diff
@@ -440,14 +463,15 @@ def load(self):
             logger.info(f"No existing DB found in {self._save_folder}, skipping load")
         else:
             path = self._save_folder + "/chroma-collections.parquet"
-            self._conn.execute(f"INSERT INTO collections SELECT * FROM read_parquet('{path}');")
+            self._conn.execute(
+                f"INSERT INTO collections SELECT * FROM read_parquet('{path}');"
+            )
             logger.info(
                 f"""loaded in {self._conn.query(f"SELECT COUNT() FROM collections").fetchall()[0][0]} collections"""
             )
 
     def __del__(self):
         logger.info("PersistentDuckDB del, about to run persist")
-        self.persist()
```

Comment:
This is the only non-linter change to duckdb.py. We remove __del__ calls to persist. 

Our use of __del__ to call persist is fundamentally wrong.

1. “It is not guaranteed that __del__() methods are called for objects that still exist when the interpreter exits.” This means that you will not reliably persist on exit. In my test it also means that the subprocess that loads and checks can do odd things. 
2. In many contexts, there are hidden references to the object that are unreliable - leading to buggy code that assumes persist on exist or that del on that object cleans it up. 
3. __del__ masks errors. As according to the documentation - dependent resources in the object may have been cleaned up in an unpredictable order. I’ve seen several transient instances of duckdb errors inside __del__ that I can’t explain but this explains it. “__del__() can be executed during interpreter shutdown. As a consequence, the global variables it needs to access (including other modules) may already have been deleted or set to None. “

Reply:
https://docs.python.org/3/reference/datamodel.html?highlight=__del__#object.__del__

Reply:
I have found an alternative to persist on write and del, which is to use https://docs.python.org/3/library/atexit.html